parquet/src/test/java/org/apache/iceberg/parquet/TestParquetUtil.java

```diff
@@ -65,6 +70,180 @@
+        required(11, "fixedCol", FixedType.ofLength(4)),
+        required(12, "binaryCol", BinaryType.get())
+    );
+    int minimumRowGroupRecordCount = 100;
+    List<Record> records = new ArrayList<>(201);
+
+    Record firstRecord = new Record(AvroSchemaUtil.convert(schema.asStruct()));
+    firstRecord.put("booleanCol", true);
+    firstRecord.put("intCol", 3);
+    firstRecord.put("longCol", 5L);
+    firstRecord.put("floatCol", 2.0F);
+    firstRecord.put("doubleCol", 2.0D);
+    firstRecord.put("decimalCol", new BigDecimal("3.50"));
+    firstRecord.put("dateCol", 1500);
+    firstRecord.put("timeCol", 2000L);
+    firstRecord.put("timestampCol", 0L);
+    firstRecord.put("uuidCol", uuid);
+    firstRecord.put("fixedCol", fixed);
+    firstRecord.put("binaryCol", "S".getBytes());
+    records.add(firstRecord);
+
+    Record secondRecord = new Record(AvroSchemaUtil.convert(schema.asStruct()));
+    secondRecord.put("booleanCol", false);
+    secondRecord.put("intCol", Integer.MIN_VALUE);
+    secondRecord.put("longCol", null);
+    secondRecord.put("floatCol", 1.0F);
+    secondRecord.put("doubleCol", null);
+    secondRecord.put("decimalCol", null);
+    secondRecord.put("dateCol", null);
+    secondRecord.put("timeCol", 3000L);
+    secondRecord.put("timestampCol", 1000L);
+    secondRecord.put("uuidCol", null);
+    secondRecord.put("fixedCol", fixed);
+    secondRecord.put("binaryCol", "W".getBytes());
+    records.add(secondRecord);
+
+    Record lastRecord = firstRecord;
+
+    for (int i = 0; i < minimumRowGroupRecordCount * 2; i++) {
+      Record newRecord = new Record(AvroSchemaUtil.convert(schema.asStruct()));
+      newRecord.put("booleanCol", true);
+      newRecord.put("intCol", Integer.valueOf(lastRecord.get("intCol").toString()) + 1);
+      newRecord.put("longCol", Long.valueOf(lastRecord.get("longCol").toString()) + 1L);
+      newRecord.put("floatCol", Float.valueOf(lastRecord.get("floatCol").toString()) + 1.0F);
+      newRecord.put("doubleCol", Float.valueOf(lastRecord.get("doubleCol").toString()) + 1.0D);
+      newRecord.put("decimalCol", new BigDecimal(lastRecord.get("decimalCol").toString())
+          .add(new BigDecimal("1.00")));
+      newRecord.put("dateCol", Integer.valueOf(lastRecord.get("dateCol").toString()) + 1);
+      newRecord.put("timeCol", Long.valueOf(lastRecord.get("timeCol").toString()) + 1L);
+      newRecord.put("timestampCol", Long.valueOf(lastRecord.get("timestampCol").toString()) + 1L);
+      newRecord.put("uuidCol", uuid);
+      newRecord.put("fixedCol", fixed);
+      newRecord.put("binaryCol", "S".getBytes());
+      records.add(newRecord);
+      lastRecord = new Record(newRecord, false);
+    }
+
+    // create parquet file with multiple row groups. by using smaller number of bytes
+    File parquetFile = writeRecords(
+        schema,
+        ImmutableMap.of(PARQUET_ROW_GROUP_SIZE_BYTES, Integer.toString(minimumRowGroupRecordCount * Integer.BYTES)),
```

Comment:
Why is this size related to the number of records? What about just using a small number, like 1mb and a few thousand records?

Reply:
@rdblue thanks for the reviewing this !

I find it is interesting :)
Yes this value can be independent, but I kept it like this, to derive number of row groups based on the rows we added. ( if we see we added 200+ records, and we got 3 row groups)

Another point I find which is interesting is about the why records has to be > 100, if we want more than 1 row groups. 
Looks like 100 is minimum value for minNumberOfRecords in each row group.
So, we use  100+ records to be written, and give rowGroupSize = (100*1 column size).

details about this value : ( I believe you would be aware of, so may be can skip lengthy part :) )
If we see this existing [TestParquet](https://github.com/apache/incubator-iceberg/blob/master/parquet/src/test/java/org/apache/iceberg/parquet/TestParquet.java#L85-L94)

It's using the 100 as the number of records, and if we run this test with < 100 records (say 80) , and even though the [here](https://github.com/apache/incubator-iceberg/blob/master/parquet/src/test/java/org/apache/iceberg/parquet/TestParquet.java#L9)1 we pass say `20` still the it doesn't create more then 1 row groups.

what I understood, reason this [startRowGroup](https://github.com/apache/incubator-iceberg/blob/master/parquet/src/main/java/org/apache/iceberg/parquet/ParquetWriter.java#L177)

It calculates this value as 
`this.nextCheckRecordCount = min(max(recordCount / 2, 100), 10000);` 
so, even if the `recordCount < 100 ` the `nextCheckRecordCount` will be 100.
and this variable is used eventually when it writes records, from [checkSize()](https://github.com/apache/incubator-iceberg/blob/master/parquet/src/main/java/org/apache/iceberg/parquet/ParquetWriter.java#L150)  which is called from [add()](https://github.com/apache/incubator-iceberg/blob/master/parquet/src/main/java/org/apache/iceberg/parquet/ParquetWriter.java#L118).

Reply:
also in the case of `testMetricsForTopLevelWithMultipleRG`, my understanding is.
it internally uses 
[ParquetWriter.java](https://github.com/apache/parquet-mr/blob/master/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetWriter.java)
which uses [InternalParquetRecordWriter.java](https://github.com/apache/parquet-mr/blob/master/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/InternalParquetRecordWriter.java#L45) which also has `MINIMUM_RECORD_COUNT_FOR_CHECK = 100` which is also making sure each row group will have minimum 100 records.